### PR TITLE
Add oss-fuzz config

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Signed-off-by: David Korczynski <david@adalogics.com>

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
@lszekeres to answer https://github.com/google/oss-fuzz/pull/8784#discussion_r1003827703

When compiling with address sanitizer, these are the flags produced:
```
build:oss-fuzz --per_file_copt=+//,-fuzztest/.*,-googletest/.*,-googlemock/.*@-O1
build:oss-fuzz --per_file_copt=+//,-fuzztest/.*,-googletest/.*,-googlemock/.*@-fno-omit-frame-pointer
build:oss-fuzz --per_file_copt=+//,-fuzztest/.*,-googletest/.*,-googlemock/.*@-gline-tables-only
build:oss-fuzz --per_file_copt=+//,-fuzztest/.*,-googletest/.*,-googlemock/.*@-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
build:oss-fuzz --per_file_copt=+//,-fuzztest/.*,-googletest/.*,-googlemock/.*@-fsanitize=address
build:oss-fuzz --per_file_copt=+//,-fuzztest/.*,-googletest/.*,-googlemock/.*@-fsanitize-address-use-after-scope
build:oss-fuzz --per_file_copt=+//,-fuzztest/.*,-googletest/.*,-googlemock/.*@-fsanitize=fuzzer-no-link
```

File: bazel/setup_configs.sh [hunk 1]
```diff
@@ -99,3 +99,54 @@ build:libfuzzer --linkopt=$(find $(${LLVM_CONFIG} --libdir) -name libclang_rt.fu
...
+### oss-fuzz compatibility mode.
+#
+# Use with: --config=oss-fuzz
+build:oss-fuzz --copt=-DFUZZTEST_COMPATIBILITY_MODE
+build:oss-fuzz --dynamic_mode=off
+build:oss-fuzz --action_env=CC=${CC}
+build:oss-fuzz --action_env=CXX=${CXX}
+"
+for flag in $CFLAGS; do
+  # whenever we have something along -fno-sanitize-recover=bool,array,...,.. we
```
Comment (lszekeres) [inline bazel/setup_configs.sh:116]: Style nit: Please start all sentences in comments with capital letters.

File: bazel/setup_configs.sh [hunk 2]
```diff
@@ -99,3 +99,54 @@ build:libfuzzer --linkopt=$(find $(${LLVM_CONFIG} --libdir) -name libclang_rt.fu
...
+  # need to split them out and write each assignment without use of commas. Otherwise
+  # the per_file_copt option splits the comma string with spaces, which causes the
+  # build command to be erroneous.
+  if [[ $flag == *","* && $flag == *"="* ]]; then
+    # Find the first occurence of equals
+    sp=(${flag//=/ })
```
Comment (lszekeres) [inline bazel/setup_configs.sh:122]: Style nit: please avoid abbreviations like `sp` and `vv`.

**Review by lszekeres — COMMENTED** [review]
Nice, thank you! And thanks for the patience!

File: bazel/setup_configs.sh [hunk 3]
```diff
@@ -99,3 +99,54 @@ build:libfuzzer --linkopt=$(find $(${LLVM_CONFIG} --libdir) -name libclang_rt.fu
 "
 
 fi # libFuzzer
+
+
+# oss-fuzz
+if [[ ${EXTRA_CONFIGS} == *"oss-fuzz"* ]]; then
```
[the remark this answers is not in the document]
Comment (DavidKorczynski) [reply bazel/setup_configs.sh]: Good idea -- fixed

**Review by lszekeres — APPROVED** [review]
@DavidKorczynski Thanks! Can I merge this, or still working on it?

**Comment by DavidKorczynski** [comment]
> @DavidKorczynski Thanks! Can I merge this, or still working on it?

Apologies for the delay. This is good to go from my side.

I've confirmed it works with https://github.com/google-research/raksha and also the example test in this repo. Tensorflow is next!

**Pull request merged**